File: java/example/src/main/java/RouteTest.java
import io.antidrift.zeromcp.*;

public class RouteTest {
    public static void main(String[] args) throws Exception {
        var server = new ZeroMcp();

        server.tool("greet", Tool.builder()
            .description("Greet a person by name")
            .input(Input.required("name", "string", "The person to greet"))
            .route("GET", "/greet/:name")
            .execute((a, ctx) -> "Hello, " + a.get("name") + "!")
            .build());

        server.tool("echo", Tool.builder()
            .description("Echo a message back")
            .input(Input.required("message", "string", "The message to echo"))
            .route("POST", "/echo")
            .execute((a, ctx) -> java.util.Map.of(
                "message", a.get("message"),
                "echoed", true
            ))
            .build());

        server.serveHttp(14256);
    }
}

File: kotlin/example/src/main/kotlin/RouteTest.kt
import io.antidrift.zeromcp.*

fun main() {
    val server = ZeroMcp()

    server.tool("greet") {
        description = "Greet a person by name"
        input {
            "name" to "string"
        }
        route("GET", "/greet/:name")
        execute { args, _ ->
            "Hello, ${args.getString("name")}!"
        }
    }

    server.tool("echo") {
        description = "Echo a message back"
        input {
            "message" to "string"
        }
        route("POST", "/echo")
        execute { args, _ ->
            mapOf(
                "message" to args.getString("message"),
                "echoed" to true
            )
        }
    }

    server.serveHttp(14257)
}
